analysis/ingest_validation.py

```diff
@@ -0,0 +1,112 @@
+from pathlib import Path
+import shutil
+
+from dcpy.utils import postgres
+from dcpy.connectors.edm import recipes
+
+from dcpy.lifecycle.ingest import run as ingest
+from dcpy.data import compare
+
+
+def compare_recipes_in_postgres(
+    dataset: str,
+    left_version: str,
+    right_version: str,
+    *,
+    build_name: str,
+    key_columns: list[str] | None = None,
+    ignore_columns: list[str] | None = None,
+    local_library_dir: str = recipes.LIBRARY_DEFAULT_PATH,
+    left_type: recipes.DatasetType = recipes.DatasetType.pg_dump,
+    right_type: recipes.DatasetType = recipes.DatasetType.pg_dump,
+):
+    ignore_columns = ignore_columns or []
+    ignore_columns.append("data_library_version")
+    left_table = dataset + "_left"
+    right_table = dataset + "_right"
+
+    client = postgres.PostgresClient(schema=build_name, database="sandbox")
+    client.drop_table(dataset)
+    client.drop_table(left_table)
+    client.drop_table(right_table)
+
+    left_ds = recipes.Dataset(id=dataset, version=left_version, file_type=left_type)
+    right_ds = recipes.Dataset(id=dataset, version=right_version, file_type=right_type)
+
+    recipes.import_dataset(
+        left_ds,
+        client,
+        import_as=left_table,
+        rename_pk=True,
+        local_library_dir=local_library_dir,
+    )
+    recipes.import_dataset(
+        right_ds,
+        client,
+        import_as=right_table,
+        rename_pk=True,
+        local_library_dir=local_library_dir,
+    )
+    if key_columns:
+        return compare.get_sql_keyed_report(
+            left_table,
+            right_table,
+            key_columns,
+            client,
+            ignore_columns=ignore_columns,
+        )
+    else:
+        return compare.get_sql_report(
+            left_table,
+            right_table,
+            client,
+            ignore_columns=ignore_columns,
+        )
+
+
+def run_ingest_and_library(
+    dataset: str,
+    ingest_parent_dir: Path = Path("."),
+    library_file_type: str = "pg_dump",
+):
+    ingest_dir = ingest_parent_dir / dataset / "special_folder"
+    print(ingest_dir)
+    ingest.run(dataset, staging_dir=ingest_dir, skip_archival=True)
+
+    from dcpy.library.archive import Archive
```

Comment:
curious why this is here?

Reply:
This doesn't really solve the issue if this is being run in any context other than a script that calls this once, as subsequent runs will have issues. But with this I was at least able to run ingest and library in one call

Reply:
Ah... gotcha. You might throw a 
```
# BEWARE: ...
```
comment on that.